src/requestLogging/requestLogging.test.ts

```diff
@@ -113,33 +169,49 @@ describe('RequestLogging', () => {
 
       const middleware = createMiddleware(logMock, headerReplacements);
 
-      const expectedError = Error('Something tragic happened');
-
-      await expect(
-        middleware(ctx, () => {
-          throw expectedError;
-        }),
-      ).rejects.toBeDefined();
-
-      expect(logMock).toHaveBeenCalledTimes(1);
-      const [logCtx, fields, error] = logMock.mock.calls[0] as unknown[];
+      await createAgent(errorHandler, middleware, handler)
+        .post('/my/failed/service')
+        .set('ACCEPT', 'image/png')
+        .set('authorization', 'retain-me')
+        .set('X-Custom-Header', 'foo')
+        .set('user-agent', 'Safari')
+        .expect(418);
 
-      expect(logCtx).toBe(ctx);
+      expect(logMock).toBeCalledTimes(1);
 
-      expect(typeof fields).toEqual('object');
+      const [, fields, err] = logMock.mock.calls[0];
 
-      const fieldsObj = fields as Record<string, unknown>;
-      expect(fieldsObj.latency).toBeGreaterThanOrEqual(0);
-      expect(fieldsObj.status).toBe(500);
-      expect(fieldsObj.internalErrorString).toBe(
-        'Error: Something tragic happened',
+      expect(fields).toMatchInlineSnapshot(
+        {
+          headers: {
+            host: expect.any(String),
+          },
+          latency: expect.any(Number),
+          'x-request-id': expect.any(String),
+        },
+        `
+        Object {
+          "headers": Object {
+            "accept": undefined,
+            "accept-encoding": "gzip, deflate",
+            "authorization": "retain-me",
```

Comment:
Why is this expected to be logged in this instance?

Reply:
Ah, added a comment to the test. A custom redaction map is used here

Reply:
oh right, ty